src/main/kotlin/com/atlassian/performance/tools/report/api/TableReporter.kt

```diff
@@ -0,0 +1,81 @@
+package com.atlassian.performance.tools.report.api
+
+import com.atlassian.performance.tools.io.api.ensureParentDirectory
+import com.atlassian.performance.tools.report.MeanAggregator
+import com.atlassian.performance.tools.report.api.result.InteractionStats
+import com.atlassian.performance.tools.report.table.AuiTabbedTableFactory
+import com.atlassian.performance.tools.report.table.AuiTable
+import com.atlassian.performance.tools.report.table.AuiTableFactory
+import org.apache.logging.log4j.LogManager
+import java.io.File
+
+class TableReporter(
+    private val output: File,
+    private val labels: List<String>
+) {
+    private val logger = LogManager.getLogger(this::class.java)
+
+    fun report(
+        data: Collection<InteractionStats>
+    ) {
+        output.ensureParentDirectory().bufferedWriter().use {
+            reportHtml(data, it)
+        }
+        logger.info("Performance results summary available at ${output.toURI()}")
+    }
+
+    private fun reportHtml(
+        data: Collection<InteractionStats>,
+        target: Appendable
+    ) {
+        val responseTimes = AuiTableFactory(
+            headers = listOf("cohort") + labels + "mean",
+            rows = data.map { stats ->
+                (stats.centers?.mapValues { (_, c) -> c.toMillis().toString() }?.toMutableMap() ?: mutableMapOf())
```

Comment:
This line is kinda busy and hard to keep track of.
I often find myself relieved when I introduce a local variable for a complex parenthesized expression.